index.js

```diff
@@ -2,57 +2,78 @@
 
 module.exports = CSSselect;
 
-var Pseudos       = require("./lib/pseudos.js"),
-    DomUtils      = require("domutils"),
-    findOne       = DomUtils.findOne,
-    findAll       = DomUtils.findAll,
-    getChildren   = DomUtils.getChildren,
-    getSiblings   = DomUtils.getSiblings,
-    removeSubsets = DomUtils.removeSubsets,
-    falseFunc     = require("boolbase").falseFunc,
-    compile       = require("./lib/compile.js"),
-    compileUnsafe = compile.compileUnsafe,
-    compileToken  = compile.compileToken;
+var DomUtils       = require("domutils"),
+    falseFunc      = require("boolbase").falseFunc,
+    compileFactory = require("./lib/compile.js"),
+    browserAdapter = require("./browser-adapter"),
+    defaultCompile = compileFactory(DomUtils);
+
+function getAdapter(options){
+	if(options && options.adapter) return options.adapter;
+
+	return DomUtils;
+}
 
 function getSelectorFunc(searchFunc){
 	return function select(query, elems, options){
+		var adapter = getAdapter(options),
+		    compile,
+		    compileUnsafe;
+
+		if(adapter === DomUtils){
+			compile = defaultCompile;
+		} else {
+			compile = compileFactory(adapter);
+		}
+
+		compileUnsafe = compile.compileUnsafe;
```

Comment:
Do you mean simply changing to something like this:

```javascript
function getSelectorFunc(searchFunc){
	return function select(query, elems, options){
		options = options || {};
		options.adapter = options.adapter || DomUtils;

		var adapter = options.adapter,
		    compile,
		    compileUnsafe;

		if(adapter === DomUtils){
			compile = defaultCompile;
		} else {
			compile = compileFactory(adapter);
		}

		compileUnsafe = compile.compileUnsafe;

		// etc.
}
```

Or were you asking about why I'm assigning the `compile` variable like that etc.? It's so that the static methods on CSSselect are always using the same instance of compile with psuedos and filters attached so that it doesn't break when the pseduos are patched at runtime by `compile.js`

Or did I misunderstand this comment entirely, haha?